File: stock_price.py
import requests
import datetime


class StockHelper(object):
    def __init__(self, stock_name='', api_key=''):
        self.stock_name = stock_name
        self.api_key = api_key


    def get_stock_data(self):
        with requests.Session() as sess_stock:
            r1 = sess_stock.get("https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol={0}&interval=1min&outputsize=compact&apikey={1}".format(self.stock_name,
                                                                                                                                                                      self.api_key))
            r1.raise_for_status()
            data = r1.json()
            return data


    def get_current_date(self):
        curr_date = datetime.datetime.now().strftime("%Y-%m-%d")
        return str(curr_date)


    def query_stock(self):
        data = self.get_stock_data()
        currDate = self.get_current_date()
        daily_data = data['Time Series (Daily)']
        if currDate in daily_data:
            dataFound = daily_data[currDate]
        print(dataFound)



if __name__ == "__main__":
    sh = StockHelper()
    sh.query_stock()

File: stock/stockprice.go
import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"net/smtp"
	"os"
	"time"
)

func (stock *StockPriceDetails) getStockData() map[string]interface{} {
	stockUrl := fmt.Sprintf("https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&"+
		"symbol=%s&interval=1min&outputsize=compact&apikey=%s",
		stock.StockName,
		stock.APIKey)

	client := &http.Client{}
	req, err := http.NewRequest("GET", stockUrl, nil)
	if err != nil {
		log.Fatal("Found Error", err)
	}
	req.Header.Set("Content-Type", "application/json")
	resp, err := client.Do(req)
	if resp.StatusCode == 200 || resp.StatusCode == 429 {
		getStockData, err := ioutil.ReadAll(resp.Body)
		if err != nil {
			log.Fatal("Found Error", err)
		}
		var stockToken StockValues
		err1 := json.Unmarshal(getStockData, &stockToken)
		if err1 != nil {
			log.Fatal("Unmarshal Error", err1)
		}
		log.Println("Stock Price Found")
		return stockToken.Token
	} else {
		log.Println("Stock Price not Found", resp)
	}

	return nil
}

func getCurrentTime() string {
	currentTime := time.Now()
	date := currentTime.Format("2006-01-02")
	fmt.Println(date)
	return date
}

func (stock *StockPriceDetails) QueryStock(timeT string) interface{} {
	stockData := stock.getStockData()
	var foundValue interface{}
	for key, value := range stockData {
		if timeT == key {
			foundValue = value
		}
	}
	return foundValue
}

func sendEmail(toEmail string, fromEmail string, emailPassword string, message string) {
	subject := "Stock Notification for Today\n\n"

	body := "From: " + fromEmail + "\n" +
		"To: " + toEmail + "\n" +
		"Subject:" + subject + message
	auth := smtp.PlainAuth("", fromEmail, emailPassword, "smtp.gmail.com")
	err := smtp.SendMail(
		"smtp.gmail.com:587",
		auth,
		fromEmail,
		[]string{toEmail},
		[]byte(body),
	)
	if err != nil {
		log.Fatal(err)
	}
}

func getEnvValue(envValue string) string {
	var valueName string
	value, con := os.LookupEnv(envValue)
	if con {
		log.Println("Found Environment Variables:", envValue)
		valueName = value
	} else {
		log.Fatalln("Value is not found", envValue)
	}
	return valueName
}